doc/examples/transform/plot_register_rotation.py

```diff
@@ -88,11 +88,120 @@
 
 print(f"Expected value for cc rotation in degrees: {angle}")
 print(f"Recovered value for cc rotation: {shiftr}")
 print()
 print(f"Expected value for scaling difference: {scale}")
 print(f"Recovered value for scaling difference: {shift_scale}")
+
+######################################################################
+# Register rotation and scaling on a translated image
+# =================================================================
+#
+# The above examples only work when the images to be registered share a
+# center. However, it is more often the case that there is also a translation
+# component to the difference between two images to be registered. One
+# approach to register rotation, scaling and translation is to first correct
+# for rotation and scaling, then solve for translation. It is possible to
+# resolve rotation and scaling differences for translated images by working on
+# the magnitude spectra of the fourier transformed images.
+
+from skimage.color import rgb2gray
+from skimage import draw, filters
+from scipy.fftpack import fft2, fftshift
+
+angle = 36
+scale = 1.4
+shiftr = 30
+shiftc = 15
+
+
+def create_image_pair(image, angle, scale, shiftr, shiftc):
+    translated = image[shiftr:, shiftc:]
+    rotated = rotate(translated, angle)
+    rescaled = rescale(rotated, scale)
+    shaper, shapec = image.shape
+    rts_image = rescaled[:shaper, :shapec]
+    return image, rts_image
+
+
+image = rgb2gray(data.retina())
+image, rts_image = create_image_pair(image, angle=angle, scale=scale,
+                                     shiftr=shiftr, shiftc=shiftc)
+
+warped_image = warp_polar(image, scaling="log")
+warped_rts = warp_polar(rts_image, scaling="log")
+
+# When center is not shared, log-polar transform is not helpful!
+fig, axes = plt.subplots(2, 2, figsize=(8, 8))
+ax = axes.ravel()
+ax[0].set_title("Original Image")
+ax[0].imshow(image)
+ax[1].set_title("Modified Image")
+ax[1].imshow(rts_image)
+ax[2].set_title("Log-Polar-Transformed Original")
+ax[2].imshow(warped_image)
+ax[3].set_title("Log-Polar-Transformed Modified")
+ax[3].imshow(warped_rts)
+plt.show()
+
+
+# Use difference of gaussians to enhance image features
+def dog(image, sigma1, sigma2):
+    image = filters.gaussian(image, sigma1) - filters.gaussian(image, sigma2)
+    return image
+
+
+image = dog(image, 5, 20)
+rts_image = dog(rts_image, 5, 20)
+
+
+# Window the images and take the magnitude of the FFT
+def window_image(image, window_diameter=0.8, window_decay=10):
+    """window_diameter is relative to length of shortest axis
+    window_decay determines steepness of window edges (higher is steeper)"""
+    window_center = np.divide(image.shape, 2)
+    radius = (window_diameter * np.min(image.shape)) / 2
+    window = np.zeros_like(image)
+    rr, cc = draw.circle(window_center[0], window_center[1], radius)
+    window[rr, cc] = 1
+    window = filters.gaussian(window, sigma=(radius / window_decay))
+    return image * window
```

Comment:
@mellertd thanks for this!

I don't have time for a full review right now but my first impression is that the example creates too many helper functions, that in fact would be useful functions to have. Specifically:

- perhaps we can add `feature.difference_of_gaussians` to the feature module (and use it within `blob_dog`), and
- windowing an image also seems like a common signal processing scenario. Are you sure that this doesn't already exist in `scipy.ndimage` or `scipy.signal`? If it doesn't, then I think we could add it to `skimage.filters`.

Reply:
> the example creates too many helper functions

That was my feeling too, but I was having trouble getting around that and maintaining (in my view) readability.

> perhaps we can add feature.difference_of_gaussians to the feature module (and use it within blob_dog)

This seems pretty easy to implement, but is it too easy? If you're just running a filter at two different sigmas and taking the difference, is that worth its own function? I'd be happy to put it together in any case.

> windowing an image also seems like a common signal processing scenario. Are you sure that this doesn't already exist in scipy.ndimage or scipy.signal? If it doesn't, then I think we could add it to skimage.filters

I am glad you think so, because--unless I am totally missing something--there is no n-dimensional windowing available in `scipy.signal`. You can take the outer product of two 1-dimensional windows for a 2D window, but that isn't exactly what you want because generally you won't get a window with circular symmetry. I'd have to do a little reading to get at the math that it would take to turn any arbitrary 1D window into a 2/3D window with circular/spherical symmetry, but I am sure I could figure it out if you want me to take a crack at it.

Reply:
> turn any arbitrary 1D window into a 2/3D window with circular/spherical symmetry

Actually, I just thought of a way this could be done with `transform.warp`. I shall sleep on this

Reply:
#4252